File: Cinema_groupe4_back/cinema_backend/src/main/java/com/intiformation/modele/Utilisateur.java
package com.intiformation.modele;

import java.sql.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Utilisateur {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id_utilisateur")
	private long id;
	private String email;
	private String motDePasse;
	private String nom;
	private String prenom;
	private String ville;
	private String numeroEtRue;
	private int codePostal;
	@Column(name = "numero_cb")
	private Long numeroCB;
	private short criptogramme;
	private Date dateNaissance;

	public Utilisateur() {
	}

	public Utilisateur(String email, String motDePasse, String nom, String prenom, String ville, String numeroEtRue,
			int codePostal, Long numeroCB, short criptogramme, Date dateNaissance) {
		this.email = email;
		this.motDePasse = motDePasse;
		this.nom = nom;
		this.prenom = prenom;
		this.ville = ville;
		this.numeroEtRue = numeroEtRue;
		this.codePostal = codePostal;
		this.numeroCB = numeroCB;
		this.criptogramme = criptogramme;
		this.dateNaissance = dateNaissance;
	}

	public Utilisateur(Long id, String email, String motDePasse, String nom, String prenom, String ville,
			String numeroEtRue, int codePostal, Long numeroCB, short criptogramme, Date dateNaissance) {
		this.id = id;
		this.email = email;
		this.motDePasse = motDePasse;
		this.nom = nom;
		this.prenom = prenom;
		this.ville = ville;
		this.numeroEtRue = numeroEtRue;
		this.codePostal = codePostal;
		this.numeroCB = numeroCB;
		this.criptogramme = criptogramme;
		this.dateNaissance = dateNaissance;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getMotDePasse() {
		return motDePasse;
	}

	public void setMotDePasse(String motDePasse) {
		this.motDePasse = motDePasse;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	public String getNumeroEtRue() {
		return numeroEtRue;
	}

	public void setNumeroEtRue(String numeroEtRue) {
		this.numeroEtRue = numeroEtRue;
	}

	public int getCodePostal() {
		return codePostal;
	}

	public void setCodePostal(int codePostal) {
		this.codePostal = codePostal;
	}

	public Long getNumeroCB() {
		return numeroCB;
	}

	public void setNumeroCB(Long numeroCB) {
		this.numeroCB = numeroCB;
	}

	public short getCriptogramme() {
		return criptogramme;
	}

	public void setCriptogramme(short criptogramme) {
		this.criptogramme = criptogramme;
	}

	public Date getDateNaissance() {
		return dateNaissance;
	}

	public void setDateNaissance(Date dateNaissance) {
		this.dateNaissance = dateNaissance;
	}

}

File: Cinema_groupe4_front/cinema_front/src/app/model/utilisateur.ts
export class Utilisateur{

    private _idUtilisateur : number
	private  _emailUtilisateur : string
	private _motDePasse : string
	private _nom : string
	private _prenom : string
	private _ville : string
	private _numeroEtRue : string
	private _codePostal : number
	private _numeroCB : number
	private _criptogramme : number
	private _dateNaissance : Date

    constructor(idUtilisateur : number, emailUtilisateur : string, motDePasse : string, nom : string, 
        prenom : string, ville : string, numeroEtRue : string, codePostal : number, numeroCB : number, 
        criptogramme : number, dateNaissance : Date){
        this._idUtilisateur =idUtilisateur
        this._emailUtilisateur = emailUtilisateur
        this._motDePasse = motDePasse
        this._nom = nom
        this._prenom = prenom
        this._ville = ville
        this._numeroEtRue = numeroEtRue
        this._codePostal = codePostal
        this._numeroCB = numeroCB
        this._criptogramme = criptogramme
        this._dateNaissance = dateNaissance
    }

    get idUtilisateur(){
        return this._idUtilisateur
    }
    set idUtilisateur (value : number){
        this._idUtilisateur = value
    }

    get emailUtilisateur(){
        return this._emailUtilisateur
    }
    set emailUtilisateur (value : string){
        this._emailUtilisateur = value
    }

    get motDePasse(){
        return this._motDePasse
    }
    set motDePasse (value : string){
        this._motDePasse = value
    }

    get nom(){
        return this._nom
    }
    set nom (value : string){
        this._nom = value
    }

    get prenom(){
        return this._prenom
    }
    set prenom (value : string){
        this._prenom = value
    }

    get ville(){
        return this._ville
    }
    set ville (value : string){
        this._ville = value
    }

    get numeroEtRue(){
        return this._numeroEtRue
    }
    set numeroEtRue (value : string){
        this._numeroEtRue = value
    }

    get codePostal(){
        return this._codePostal
    }
    set codePostal (value : number){
        this._codePostal = value
    }

    get numeroCB(){
        return this._numeroCB
    }
    set numeroCB (value : number){
        this._numeroCB = value
    }
    
    get criptogramme(){
        return this._criptogramme
    }
    set criptogramme (value : number){
        this._criptogramme = value
    }

    get dateNaissance(){
        return this._dateNaissance
    }
    set dateNaissance (value : Date){
        this._dateNaissance = value
    }


}